script/scaffold/templates/config_flow/integration/config_flow.py

```diff
@@ -90,8 +90,11 @@ async def async_step_user(
             else:
                 return self.async_create_entry(title=info["title"], data=user_input)
 
+        data_schema = self.add_suggested_values_to_schema(
+            STEP_USER_DATA_SCHEMA, user_input
+        )
         return self.async_show_form(
-            step_id="user", data_schema=STEP_USER_DATA_SCHEMA, errors=errors
+            step_id="user", data_schema=data_schema, errors=errors
```

Comment:
I don't think this fits.

This is mostly not needed to add, nor is it used or useful in the existing set example in this scaffold.

Reply:
In the context of the example, it means that you don't need to type in the username again if the password was wrong.

From a UI perspective I think it's a huge improvement and likely to be so on pretty much every config flow.

Reply:
I could add a comment to make that clearer...

Reply:
I don't agree.

If anything, IMHO, the scaffolding already has too much-opinionated logic, making it more difficult to start and just raising the bar.